packages/react/menu/src/Menu.tsx

```diff
@@ -435,16 +437,62 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
     ...contentProps
   } = props;
   const context = useMenuContext(CONTENT_NAME);
-  const typeaheadProps = useMenuTypeahead();
   const { getItems } = useCollection();
   const [currentItemId, setCurrentItemId] = React.useState<string | null>(null);
   const contentRef = React.useRef<HTMLDivElement>(null);
   const composedRefs = useComposedRefs(forwardedRef, contentRef, context.onContentChange);
   const isPointerDownOutsideRef = React.useRef(false);
+  const timerRef = React.useRef(0);
+  const searchRef = React.useRef('');
+  const [isSearching, setIsSearching] = React.useState(false);
 
   const PortalWrapper = portalled ? Portal : React.Fragment;
   const ScrollLockWrapper = disableOutsideScroll ? RemoveScroll : React.Fragment;
 
+  // Reset `searchRef` 1 second after it was last updated
+  const setSearch = React.useCallback((search: string) => {
+    searchRef.current = search;
+    setIsSearching(search.length > 0);
+    window.clearTimeout(timerRef.current);
+    timerRef.current = window.setTimeout(() => {
+      setSearch('');
+    }, 1000);
```

Comment:
nit 😛 

```suggestion
    timerRef.current = window.setTimeout(() => setSearch(''), 1000);
```

haha, me and my wrapping woes!